draft-ietf-acme-acme.md

```diff
@@ -1419,8 +1415,63 @@ status of the order will indicate what action the client should take:
   is in the process of generating the certificate.  Retry after the time given
   in the "Retry-After" header field of the response, if any.
 
-* "valid": The server has issued the certificate and provisioned its URL to the
-  "certificate" field of the order.
+* "valid": The server has agreed to issue the certificate; it can be fetched
+  from the URL in the "certificate" field of the order
+
+
+### Requiring Authorization before Order
+
+In some cases, a server may require a client to complete the authorizations
+necessary to issue a certificate before it will accept a new-order request.
+This allows the server to avoid caching a CSR before authorizations have been
+completed.  When such a server receives a new-order request without sufficient
+authorizations, it should return an error response with status code 401
+(Unauthorized) and error code "authorizationFirst".  Any error document with
+this error code MUST include a field "authorizations" containing an array of
+URLs for authorization objects that must be completed before the new-order
+request can succeed.
```

Comment:
This seems awkward - are there any other places in the protocol where the caller is expected to pull a non-standard field out of an error document in order to learn how to proceed with a subsequent API operation?

Reply:
Don't know if this counts as a subsequent API operation, but the [section on re-agreement to ToS](https://ietf-wg-acme.github.io/acme/draft-ietf-acme-acme.html#rfc.section.7.3.4) has the client direct the user to a URI from the "instance" field.

In any case, the inclusion of additional data in problem documents is [well defined](https://tools.ietf.org/html/rfc7807#section-3.2).